src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Caching.cs

```diff
@@ -252,52 +257,114 @@ public static CachingContext GetOrCreate(JsonSerializerOptions options)
+                        return false;
+                    }
+
+                    for (int i = 0; i < n; i++)
+                    {
+                        if (left[i] != right[i])
+                        {
+                            return false;
+                        }
+                    }
+
+                    return true;
+                }
+            }
+
+            public int GetHashCode(JsonSerializerOptions options)
             {
-                int n;
-                if ((n = left.Count) != right.Count)
+                HashCode hc = default;
+
+                AddHashCode(ref hc, options._dictionaryKeyPolicy);
+                AddHashCode(ref hc, options._jsonPropertyNamingPolicy);
+                AddHashCode(ref hc, options._readCommentHandling);
+                AddHashCode(ref hc, options._referenceHandler);
+                AddHashCode(ref hc, options._encoder);
+                AddHashCode(ref hc, options._defaultIgnoreCondition);
+                AddHashCode(ref hc, options._numberHandling);
+                AddHashCode(ref hc, options._unknownTypeHandling);
+                AddHashCode(ref hc, options._defaultBufferSize);
+                AddHashCode(ref hc, options._maxDepth);
+                AddHashCode(ref hc, options._allowTrailingCommas);
+                AddHashCode(ref hc, options._ignoreNullValues);
+                AddHashCode(ref hc, options._ignoreReadOnlyProperties);
+                AddHashCode(ref hc, options._ignoreReadonlyFields);
+                AddHashCode(ref hc, options._includeFields);
+                AddHashCode(ref hc, options._propertyNameCaseInsensitive);
+                AddHashCode(ref hc, options._writeIndented);
+                AddHashCode(ref hc, options._typeInfoResolver);
+                AddListHashCode(ref hc, options._converters);
+
+                return hc.ToHashCode();
+
+                static void AddListHashCode<TValue>(ref HashCode hc, ConfigurationList<TValue> list)
                 {
-                    return false;
+                    int n = list.Count;
+                    for (int i = 0; i < n; i++)
+                    {
+                        AddHashCode(ref hc, list[i]);
+                    }
                 }
 
-                for (int i = 0; i < n; i++)
+                static void AddHashCode<TValue>(ref HashCode hc, TValue? value)
                 {
-                    TValue? leftElem = left[i];
-                    TValue? rightElem = right[i];
-                    bool areEqual = leftElem is null ? rightElem is null : leftElem.Equals(rightElem);
-                    if (!areEqual)
+                    if (typeof(TValue).IsValueType)
                     {
-                        return false;
+                        hc.Add(value);
+                    }
+                    else
+                    {
+                        Debug.Assert(!typeof(TValue).IsSealed, "Sealed reference types like string should not use this method.");
```

Comment:
Can user defined convertors show up here? What prevents them from being sealed?

Reply:
They can, but the static type is JsonConverter which is not sealed.

Reply:
Ah ok